### PR TITLE
Pinch swipe some fixs

### PR DESCRIPTION
**Pull request by webcarrot**
Add pinch to zoom and swipe.

Support pointer/touch/mouse events.

Tested on iPad (safari), win10 tablet (chrome, edge) + desktop.

Should works with android browsers.

PS. Rotation in future.

### CODE REVIEW

File: src/react-image-lightbox.js [hunk 1]
```diff
@@ -168,20 +209,46 @@ class ReactImageLightbox extends Component {
...
+        const id = setTimeout(() => {
+            this.timeouts = this.timeouts.filter(tid => tid !== id);
+            func();
+        }, time);
+        this.timeouts.push(id);
+        return id;
+    }
+
+    clearTimeout(id) {
+        if (this.timeouts.find(tid => tid === id)) {
```
Comment (fritz-c) [inline src/react-image-lightbox.js]: [`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Browser_compatibility) isn't supported in IE.

**Comment by fritz-c** [comment]
This is just __awesome__.

I saw the subject line in my inbox this morning and thought someone had yet again made a new issue requesting this essential functionality. Going once more over all the unknowns of implementing this in my head, I felt weighted down thinking about how long it would take me to get it working.
So imagine my surprise when I found out it was not an issue, but a _pull request!_

I tried to avoid becoming too optimistic before trying it out for myself, because I thought that even if the touch functionality worked, there was a good chance it would break more than it fixed. But it didn't! It works great!

Now I just have to go through the motions of closing all the issues opened before about this, mark it off on the milestones, tell all the library users that now they can do what they had been requesting they be able to do ever since I released this library.

Thank you so much.

**Comment by fritz-c** [comment]
It's now available in `v3.4.0`. Thanks again!

By the way, what do you mean by rotation (in your 'Rotation in future' comment up there)?

**Comment by webcarrot** [comment]
I can quite easily add support for rotation around image center. ( 3+ pointers and buttons as fallback).

It is more difficult ( but better ) to do the rotation around the center designated by the pointers ( probably we need to manipulate transform-origin and/or translate).

I think it could be useful in some cases - for example: page with images uploaded by "random" users.

Rotation could be free or semi free.

If you like this idea: 👍 if not: 👎

PS. Sorry for my English.

**Comment by fritz-c** [comment]
I'm going to say 👎  just because I think the number of cases where it would be useful is quite small.

**Comment by webcarrot** [comment]
I think similar. 🙂

**Pull request merged**